packages/appium/lib/extension/plugin-config.js

```diff
@@ -0,0 +1,111 @@
+// @ts-check
+
+import _ from 'lodash';
+import {ExtensionConfig} from './extension-config';
+import log from '../logger';
+import {PLUGIN_TYPE} from '../constants';
+
+/**
+ * @extends {ExtensionConfig<PluginType>}
+ */
+export class PluginConfig extends ExtensionConfig {
+
+  /**
+   * A mapping of {@link Manifest} instances to {@link PluginConfig} instances.
+   *
+   * `Manifest` and {@link ExtensionConfig} have a one-to-many relationship; each `Manifest` should be associated with a `DriverConfig` and a `PluginConfig`; no more, no less.
```

Comment:
It's...well, that depends if you would define "two" as "many".

Reply:
it just made it hard for me to grok this comment, as if there was more structure possible here than there is. it would have read more straightforward to me as something like just: "Each `Manifest` has two  `ExtensionConfig`s; one for plugins and one for drivers"